File: app/src/main/java/wangdaye/com/geometricweather/db/dao/LocationDao.java
package wangdaye.com.geometricweather.db.dao;

import androidx.annotation.Nullable;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import wangdaye.com.geometricweather.db.entities.LocationEntity;

@Dao
public interface LocationDao {

    @Insert
    void insertLocationEntity(LocationEntity entity);

    @Insert
    void insertLocationEntityList(List<LocationEntity> entityList);

    @Delete
    void deleteLocationEntity(LocationEntity entity);

    @Query("DELETE FROM LOCATION_ENTITY")
    void deleteLocationEntityList();

    @Update
    void updateLocationEntity(LocationEntity entity);

    @Query("SELECT * FROM LOCATION_ENTITY WHERE FORMATTED_ID = :formattedId LIMIT 1")
    @Nullable
    LocationEntity selectLocationEntity(String formattedId);

    @Query("SELECT * FROM LOCATION_ENTITY")
    List<LocationEntity> selectLocationEntityList();

    @Query("SELECT COUNT(*) FROM LOCATION_ENTITY")
    int countLocationEntity();
}

File: app/src/main/java/wangdaye/com/geometricweather/db/dao/LocationDao.kt
package wangdaye.com.geometricweather.db.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import wangdaye.com.geometricweather.db.entities.LocationEntity

@Dao
interface LocationDao {

    @Insert
    fun insertLocationEntity(entity: LocationEntity)

    @Insert
    fun insertLocationEntityList(entityList: List<LocationEntity>)

    @Delete
    fun deleteLocationEntity(entity: LocationEntity)

    @Query("DELETE FROM LOCATION_ENTITY")
    fun deleteLocationEntityList()

    @Update
    fun updateLocationEntity(entity: LocationEntity)

    @Query("SELECT * FROM LOCATION_ENTITY WHERE FORMATTED_ID = :formattedId LIMIT 1")
    fun selectLocationEntity(formattedId: String): LocationEntity?

    @Query("SELECT * FROM LOCATION_ENTITY")
    fun selectLocationEntityList(): List<LocationEntity>

    @Query("SELECT COUNT(*) FROM LOCATION_ENTITY")
    fun countLocationEntity(): Int
}
